src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -1234,6 +1437,85 @@ int fx_muxer_t::execute(
     return result;
 }
 
+int fx_muxer_t::get_coreclr_delegate(
+    const host_startup_info_t &host_info,
+    coreclr_delegate_type type,
+    void **delegate)
+{
+    assert(host_info.is_valid());
+
+    const host_mode_t mode = host_mode_t::libhost;
+
+    // Read config
+    fx_definition_vector_t fx_definitions;
+    auto app = new fx_definition_t();
+    fx_definitions.push_back(std::unique_ptr<fx_definition_t>(app));
+
+    pal::string_t runtime_config = _X("");
+    fx_reference_t override_settings;
+    int rc = read_config(*app, host_info.app_path, runtime_config, override_settings);
+    if (rc != StatusCode::Success)
```

Comment:
Is there a need to cache this results?  Creating a lot of delegates could make this code path really hot.

Reply:
100 % agreed. You have touched on a pontentially idempotent optimization, but it may only be idempotent based on how we define previously loaded coreclr instances and how that equivalency is defined.